### DISCUSSION TITLE
Keda ScaledJob with Sidekiq

### DISCUSSION THREAD

**Discussion by blaizmannen** [kedacore/keda#4051] [Q&A / Need Help]
So we used to run our sidekiq(background/worker processing for ruby apps) as deployments in our k8s cluster. Recently we've been having problems of having our sidekiq pods having long running tasks being terminated due to scaling events/helm upgrades. One way to solve this is to increase the `terminationGracePeriodSeconds` to a high value (example a few hours). However, this will block new deployments of our application.

We have been exploring the use of KEDA `ScaledJob` to handle the scaling of our sidekiq pods and also prevent long running tasks from blocking new sidekiq deployment upgrades.

Converting the Sidekiq deployment to a ScaledJob has worked to some extent and it is able to process tasks. However, the pod does not exit after tasks are being run, and will only terminate if an  `activeDeadlineSeconds` config is set.  This is because our spec for the Job is made for that of a deployment, not a job that will exit under a certain condition.

Would anyone have a suggestion on how to design our sidekiq applications so that KEDA can scale the sidekiq jobs on the fly, and exit accordingly when they are done? How do we determine that a sidekiq job is ready to terminate and that it will not affect newly incoming sidekiq jobs?

Do let me know if there is any more info I can provide as the above writeup is very high level.

**Comment by tomkerkhove** [comment]
I'm not aware of what sidekiq is or how it works, but can't the app shut down when the work is done? Because that would solve your problem.

**Reply by pmquang** [reply]
the thing is sidekiq just like other normal applications, a long running app, it's not supposed to be shutdown when it done processing 1 message. It keeps pulling new messages to continue to process.
